pilot/pkg/model/push_context.go

```diff
@@ -1286,8 +1288,8 @@ func (ps *PushContext) updateContext(
 		ps.serviceAccounts = oldPushContext.serviceAccounts
 	}
 
-	if servicesChanged || gatewayAPIChanged {
-		// Gateway status depends on services, so recompute if they change as well
+	if servicesChanged || gatewayAPIChanged || secretsChanged {
```

Comment:
This is starting to push the limits of our abstractions a bit.

So lets say a secret is added.

The credentials controller triggers `Push{full: false, kind: secret}`. This skips psuh context update; all the XDS types other than SDS skip it; SDS pushes.

Next Gateway controller triggers `Push{full: true, kind: secret}`. This hits this code, we re-init the gateway context, which triggers a status update. This _should_ have no impact on XDS! But due to the coupling, we end up doing a real SDS push (again)

In some case, debounce may join these anyways though

Reply:
Yeah I have the same concern, 2 possible options I thought of:
1. Switch `ConfigsUpdated.Kind` from `Secret` to `KubernetesGateway`, then it will not mess up with SDS pushes. The problem is that we don't know the exact namespace/name of the impacted Gateway, so we have to set an incomplete updated config, which might be risky as @hzxuzhonghu mentioned [here](https://github.com/istio/istio/pull/41920#discussion_r1019798346)
2. Directly handle secret events in the Gateway controller without a full push, then we may need to consider things like workqueue, debounce, concurrency control..

Reply:
Tried (1) with finding out the impacted gateways